### DISCUSSION TITLE
Wasm Application Close

### DISCUSSION THREAD

**Discussion by adampulley** [unoplatform/uno#9667] [Q&A] [answered]
So a Wasm app sadly doesn't get a CloseRequested call when a user closes their browser tab. So instead I want to add a close button to my app to allow me to tidy up and then cleanly close things and then finally close the Uno app, ideally causing the browser tab to then close too. However I can't seem to find a function to call to cause the close - is there one?

Windows.ApplicationModel.Core.CoreApplication.Exit(); isn't supported ("is not implemented in Uno")

**Comment by FrozDark** [comment]
You can't force user to close the tab or browser through js or wasm (security). You can show some page instead

https://stackoverflow.com/questions/55603206/how-to-exit-from-a-webassembly-process-on-the-front-end

**Comment by jeromelaban** [comment, marked as answer]
See https://github.com/unoplatform/uno/discussions/9596 for WebAssembly exit support. 

The `CloseRequested` event is not implemented, but you can use [Suspending event](https://platform.uno/docs/articles/features/windows-ui-xaml-application.html?q=suspending#applicationsuspending-event) to some extents.

**Comment by adampulley** [comment]
Thank you for the replies, makes sense and I'll give that event a try thanks
